redisvl/extensions/session_manager/semantic_session.py

```diff
@@ -0,0 +1,365 @@
+
+
+class SemanticSessionManager(BaseSessionManager):
+    session_field_name: str = "session_tag"
+    user_field_name: str = "user_tag"
+    vector_field_name: str = "vector_field"
+
+    def __init__(
+        self,
+        name: str,
+        session_tag: str,
+        user_tag: str,
+        prefix: Optional[str] = None,
+        vectorizer: Optional[BaseVectorizer] = None,
+        distance_threshold: float = 0.3,
+        redis_client: Optional[Redis] = None,
+    ):
+        """Initialize session memory with index
+
+        Session Manager stores the current and previous user text prompts and
+        LLM responses to allow for enriching future prompts with session
+        context. Session history is stored in prompt:response pairs referred to
+        as exchanges.
+
+        Args:
+            name (str): The name of the session manager index.
+            session_tag (str): Tag to be added to entries to link to a specific
+                session.
+            user_tag (str): Tag to be added to entries to link to a specific user.
+            prefix (Optional[str]): Prefix for the keys for this session data.
+                Defaults to None and will be replaced with the index name.
+            vectorizer (Vectorizer): The vectorizer to create embeddings with.
+            distance_threshold (float): The maximum semantic distance to be
+                included in the context. Defaults to 0.3.
+            redis_client (Optional[Redis]): A Redis client instance. Defaults to
+                None.
+
+        The proposed schema will support a single vector embedding constructed
+        from either the prompt or response in a single string.
+
+        """
+        super().__init__(name, session_tag, user_tag)
+
+        prefix = prefix or name
+
+        self._vectorizer = vectorizer or HFTextVectorizer(
+            model="sentence-transformers/msmarco-distilbert-cos-v5"
+        )
+
+        self.set_distance_threshold(distance_threshold)
+
+        schema = IndexSchema.from_dict({"index": {"name": name, "prefix": prefix}})
+
+        schema.add_fields(
+            [
+                {"name": "role", "type": "text"},
+                {"name": "content", "type": "text"},
+                {"name": "tool_call_id", "type": "text"},
+                {"name": "timestamp", "type": "numeric"},
+                {"name": "session_tag", "type": "tag"},
+                {"name": "user_tag", "type": "tag"},
+                {
+                    "name": "vector_field",
+                    "type": "vector",
+                    "attrs": {
+                        "dims": self._vectorizer.dims,
+                        "datatype": "float32",
+                        "distance_metric": "cosine",
+                        "algorithm": "flat",
+                    },
+                },
+            ]
+        )
+
+        self._index = SearchIndex(schema=schema)
+
+        if redis_client:
```

Comment:
Not a blocker just thinking out loud. 

What are the advantages of passing a redis_client vs just a redis_url? If I'm a user I think I'd rather just provide the connection string and let the class do the rest. Then this code gets simplier because it's just redis_url or "redis://localhost:6379" 

I'm also wondering if we should overwrite by default in case of collision issues... but could go either way.